flake-modules/dev/default.nix

```diff
@@ -66,6 +66,15 @@
         check.enable = false;
 
         settings.hooks = {
+          deadnix = {
+            enable = true;
+
+            settings = {
+              noLambdaArg = true;
+              noLambdaPatternNames = true;
+              edit = true;
+            };
+          };
```

Comment:
I'm on the fence whether this is good as a pre-commit hook. This could be frustrating while working on unfinished PRs.

My instinct is that this may be better as a separate flake check and/or a _github_ workflow.

The benefit of a github workflow over buildbot here is that it can provide more integrated feedback inline within PRs and be more easily distinguished from other failures.

That said, I'm fine to merge this and later refactor/revert if needed.

Reply:
That's fair, I think with these flags it shouldn't be too disruptive. But, if we all start to find it annoying we can pivot pretty easily, at least.